app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/server/WebRtcManager.kt

```diff
@@ -77,22 +84,54 @@ class WebRtcManager constructor(
         audioSource = peerConnectionFactory.createAudioSource(MediaConstraints())
         audioTrack = peerConnectionFactory.createAudioTrack("audio", audioSource)
         cameraVideoCapturer?.let { enableVideo(true, it) }
-        rtcServerConnectionObserver = RtcServerConnectionObserver()
+        createConnection()
+    }
+
+    private fun createConnection() {
+        connectionObserver = ConnectionObserver()
 
         peerConnection = peerConnectionFactory.createPeerConnection(
             emptyList(),
-            rtcServerConnectionObserver
+            connectionObserver
         )
-
         val stream = peerConnectionFactory.createLocalMediaStream("stream")
         stream.addTrack(audioTrack)
         stream.addTrack(videoTrack)
         peerConnection?.addStream(stream)
+
+        listenForIceCandidates(connectionObserver.streamObservable)
+    }
+
+    private fun listenForIceCandidates(streamObservable: Observable<StreamState>) {
+        streamObservable
+            .subscribeOn(Schedulers.io())
+            .ofType(OnIceCandidatesChange::class.java)
+            .subscribeBy(onNext = { iceCandidateChange ->
+                when (iceCandidateChange.iceCandidateState) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/8ce941c6c7cc4d13bad0e73c61419fa5d1ca7b0f/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/server/WebRtcManager.kt#L110"></span>Prefer using ‘if’ for binary conditions instead of ‘when’.</td>
    </tr>
  </tbody>
</table>